### PR TITLE
fix: update eslint-plugin-sonarjs to latest version 0.21.0

### LINKED ISSUE #13: Upgrade the eslint-plugin-sonarjs to ^0.21.0
**Is your feature request related to a problem? Please describe.**
Currently the `eslint-plugin-sonarjs` is ^0.20.0

**Describe the solution you'd like**
Need to upgrade to ^0.21.0 or above. See [changelog](https://github.com/SonarSource/eslint-plugin-sonarjs/releases).

**Additional context**
See the contributing [guide](https://github.com/dipiash/eslint-plugin-nimbus-clean/blob/main/CONTRIBUTING.md).

### PR DESCRIPTION
**Pull request by areebniyas**
Closes #13

### CODE REVIEW

**Comment by areebniyas** [comment]
Hi @dipiash, can I get some feedback on this?

**Review by dipiash — APPROVED** [review]

**Comment by dipiash** [comment]
Hi, @areebniyas!

Thank you for your improvements in this report

File: package.json [hunk 1]
```diff
@@ -88,7 +88,7 @@
     "eslint-plugin-react": "^7.33.2",
     "eslint-plugin-react-hooks": "^4.6.0",
     "eslint-plugin-simple-import-sort": "^10.0.0",
-    "eslint-plugin-sonarjs": "^0.20.0",
+    "eslint-plugin-sonarjs": "^0.21.0",
```
[the remark this answers is not in the document]
Comment (areebniyas) [reply package.json:91]: Hi, could we run the workflow now.

Also, best practices include having `package-lock.json` file in the `.gitignore`. What do you think of that suggestion? Including the `package-lock.json` in the `.gitignore` file?

**Comment by areebniyas** [comment]
@dipiash Could we run the workflow now?

Also, best practices include having package-lock.json file in the .gitignore. What do you think of that suggestion? Including the package-lock.json in the .gitignore file?

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (areebniyas) [reply package.json:91]: Fixed with [`9b45b56`](https://github.com/dipiash/eslint-plugin-nimbus-clean/pull/17/commits/9b45b562526a28994875be5711f0a024dea196b8)

**Comment by dipiash** [comment]
> @dipiash Could we run the workflow now?
> 
> Also, best practices include having `package-lock.json` file in the `.gitignore`. What do you think of that suggestion? Including the `package-lock.json` in the `.gitignore` file?

I think that keep `package-lock.json` in repository is a good choice because:
- ensures that all contributors are using the exact same dependency versions. This can help prevent inconsistencies and bugs caused by different versions of dependencies
- guarantees reproducibility of the project's environment

And in my mind If don't keep the file in repository it's may encounter difficulties reproducing the exact development environment, especially when setting up the project on a different machine or at a later date

**Review by areebniyas — COMMENTED** [review]
Fixed with [`9b45b56`](https://github.com/dipiash/eslint-plugin-nimbus-clean/pull/17/commits/9b45b562526a28994875be5711f0a024dea196b8)

**Comment by dipiash** [comment]
Thank you 👍

**Pull request merged**